src/components/single-create.ts

```diff
@@ -0,0 +1,138 @@
+    
+    const vectorLayer = new VectorLayer({
+      source: vectorSource,
+      visible: true,
+      style: [CreateStyle.setupSingleClick(1), CreateStyle.setupSingleClickCenterCircle(1)],
+    });
+
+    map.addLayer(vectorLayer);
+
+    map.getView().on('change:resolution', () => {
+      const zoom = map.getView().getZoom();
+      const resolution = map.getView().getResolution();
+      if (zoom && resolution && zoom > minZoomAllowed ) vectorLayer.setStyle([CreateStyle.setupSingleClick(zoom / resolution), CreateStyle.setupSingleClickCenterCircle(zoom / resolution)])
+    })
+  }
+
+  setupCreateElement(map: Map, vectorSource:Vector) {
+    const feature = useStore().getSelectedFeature();
+    if (feature) {
+      if (this.currentFeature) {
+        vectorSource.removeFeature(this.currentFeature)
+        map.getControls().forEach((control: Control) => {
+          if (control instanceof SelectCreateInformationBoxController) {
+            map.removeControl(control);
+          }
+        });
+      }
+      this.currentFeature = feature;
+      vectorSource.addFeature(this.currentFeature);
+      map.addControl(new SelectCreateInformationBoxController(feature.get('geometry').getCoordinates()));
+      useStore().setCustomDisplay(true);
+      useStore().setTargetBoxSize('select');
+    }
+    useStore().getMap().get('target').className = `${useStore().getTargetBoxSize()} ${useStore().getTheme()}`
+  }
+
+  setupDeleteElement(map: Map, vectorSource:Vector) {
+    if (this.currentFeature) {
+      vectorSource.removeFeature(this.currentFeature)
+        map.getControls().forEach((control: Control) => {
+          if (control instanceof SelectCreateInformationBoxController) {
+            map.removeControl(control);
+          }
+        });
+      vectorSource.removeFeature(this.currentFeature);
+      this.currentFeature = undefined;
+      useStore().setCustomDisplay(false);
+      useStore().setTargetBoxSize('no-box');
+    }
+    useStore().getMap().get('target').className = `${useStore().getTargetBoxSize()} ${useStore().getTheme()}`
+  }
+
+  longClickEvent(mapElement: HTMLDivElement, map: Map) {
+    const longClickDuration = 800;
+    let startTime: number, endTime: number;
+    let longpress: boolean = false;
+
+    // Desktop device
+    // pointerdown and pointerup aren't supported by openlayers map
+    mapElement.addEventListener('pointerdown', () => {
+      startTime = new Date().getTime();
+    });
+    mapElement.addEventListener('pointerup', () => {
+      endTime = new Date().getTime();
+      longpress = endTime - startTime < longClickDuration ? false : true;
+    });
+
+    map.on('click',  (e) => {
+      if (longpress) {
+        const geomPoint = new Point(e.coordinate);
+        const feature = new Feature({
+          geometry: geomPoint,
+        });
+        useStore().setSelectedFeature(feature)
+        GeocityEvent.sendEvent('icon-created', undefined);
+      }
+    });
+
```

Comment:
the interaction  is not really intuitive. The point should be created as soon as the mouse is pressed for  `longClickDuration` and should not wait for the user to unpress the mouse

Reply:
Refactoring done